ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -224,8 +224,27 @@ static zend_string* sqlite_handle_quoter(pdo_dbh_t *dbh, const zend_string *unqu
 {
 	char *quoted;
 	if (ZSTR_LEN(unquoted) > (INT_MAX - 3) / 2) {
+		php_error_docref(NULL, E_WARNING, "String is too long to be quoted");
 		return NULL;
 	}
+	// check if it contains a null byte
+	if (memchr(ZSTR_VAL(unquoted), '\0', ZSTR_LEN(unquoted))) {
+		if(((ZSTR_LEN(unquoted) * 2) + 3) > (INT_MAX - 3) / 2) {
+			php_error_docref(NULL, E_WARNING, "Binary string is too long to be quoted");
+			return NULL;
+		}
+		zend_string *quoted_str = zend_string_alloc((ZSTR_LEN(unquoted) * 2) + 4, 0); // 4: x'hex'\x00
+		// x'hex'
+		quoted_str->val[0] = 'x';
```

Comment:
Don't open-code `->val` and `->len`, use the `ZSTR_VAL` and `ZSTR_LEN` macros, which will be robust if the data structure ever changes.